### ISSUE TITLE
Improve dialog box message for failed pull-rebase when locally modified files and unpushed commits exist

### ISSUE THREAD

**Issue by chrispy-snps** [oxygenxml/oxygen-git-client-addon#173]
When all of the following are true:

* There are locally modified files (staged or unstaged)
* There are unpulled upstream commits
* There are unpushed local commits

then `git pull --rebase` returns the following:

```
$ git pull --rebase
error: cannot pull with rebase: You have unstaged changes.
error: please commit or stash them.
```

The equivalent Git add-on dialog box in Oxygen is as follows:

![image](https://user-images.githubusercontent.com/50950969/214441770-e05cdeeb-d0f0-49dd-a5b6-2cb67cc926ef.png)

Sometimes writers interpret this message to mean that the listed files conflict, because this dialog looks similar to the add-on's conflicting files dialog box. This sometimes confuses our writers because they will change a file, then get this message, then wonder why someone else is also working on their content.

However, in this case, Git cannot perform the rebase process any any locally modified files, even if they are completely unaffected by the incoming changes. They simply prevent the rebase process from being able to temporarily walk the local worktree through the past commits to rebase them. (In the example above, note that the incoming commit modifies "test.dita" that is not locally modified.)

Perhaps the add-on's dialog text could better explain what Git's actual requirements are:

> Cannot pull (rebase) because there are local uncommitted changes to the files below. (The incoming remote changes might not modify these files, but unpushed commits cannot be rebased when local changes also exist.)

**Comment by sorincarbunaru** [comment]
Note: internal issue is EXM-52760.

**Comment by chrispy-snps** [comment]
@AlexSmarandache15 - thanks for having a look at this!

I put all of the alternatives together, then converged upon the following text:

* Cannot pull (rebase) because there are uncommitted local changes to the files below. The incoming remote changes might not modify these files, but a rebase cannot be performed when uncommitted local changes exist.
  Commit, stash, or discard the local changes and try again.

I believe this most accurately describes the underlying limitation of Git. Are you agreeable to this text?

**Comment by AlexSmarandache15** [comment]
@chrispy-snps Thanks! It looks good to me.

**Comment by AlexJitianu** [comment]
We released version 5.2.0 which contains the improved message.

**Comment by chrispy-snps** [comment]
Thank you @AlexJitianu!
